docs/design/references/METADATA.md

```diff
@@ -0,0 +1,25 @@
+# `nupm`'s package metadata reference
+
+## Required attributes
+- `name`: Package name
+- `nu-version`: Supported Nushell version. Just like `dependencies` might be either exact version or some matcher like "greater than 1.70"
+- `version`: Version of the package. Semantic versioning is advised to enable users to have more generic requirements
+- `type`: Tells `nupm` how to install the package
+
+## Required attributes for publishing
+- `author`: Name of the developer/organization/etc.
+- `short-description`: Short info about the package, displayed by default
+- `supported-os`: Operating systems supported by the package, the most broad possibility: `{"arch": ["*"], "family": ["*"], "name": ["*"]}`. Matched by `$nu.os-info`
```

Comment:
I disagree with marking it `for publishing`. We should always validate if the package is actually installable on the current host.

Reply:
I'm thinking about a scenario where you start with `nupm new` and install it only locally: You write it on your machine so it implicitly supports your OS, you don't need to specify that. Looking at `cargo init` as a reference, it also has only three fields. Also, Nushell is cross-platform, we should encourage that by assuming project is cross-platform. If your project is not cross-platform, then you can fill in the `supported-os`.